src/components/interaction-designer/block-editors/choices/SelectOneResponseBlockContactPropertyEditor.vue

```diff
@@ -0,0 +1,178 @@
+                :options="choiceValueOptions"
+                :show-labels="false"
+                :placeholder="trans('flow-builder.select-a-value')"
+                track-by="value"
+                label="description"
+                @input="setChoiceValueOption($event, choiceKey)" />
+              <input
+                v-if="isTextProperty"
+                :value="getChoiceValue(choiceKey)"
+                type="text"
+                class="form-control"
+                :placeholder="trans('flow-builder.enter-value')"
+                @input="setChoiceValue($event.target.value, choiceKey)">
+              <input
+                v-if="isNumberProperty"
+                :value="getChoiceValue(choiceKey)"
+                type="number"
+                class="form-control"
+                :placeholder="trans('flow-builder.enter-value')"
+                @input="setChoiceValue(Number($event.target.value), choiceKey)">
+            </div>
+          </template>
+        </div>
+      </template>
+    </generic-contact-property-editor>
+  </div>
+</template>
+
+<script lang="ts">
+import VueMultiselect from 'vue-multiselect'
+import Component, {mixins} from 'vue-class-component'
+import {Prop} from 'vue-property-decorator'
+import Lang from '@/lib/filters/lang'
+import {ISelectOneResponseBlock} from '@floip/flow-runner/src/model/block/ISelectOneResponseBlock'
+import {IContactPropertyOption} from '@/store/flow/block-types'
+import {Getter, namespace} from 'vuex-class'
+import {find} from 'lodash'
+import {IContactPropertyMultipleChoice} from '@/store/flow/block-types/Core_SetContactPropertyStore'
+import {choicesToExpression} from './expressionTransformers'
+
+const flowVuexNamespace = namespace('flow')
+
+@Component({
+  components: {
+    VueMultiselect,
+  },
+})
+export class SelectOneResponseBlockContactPropertyEditor extends mixins(Lang) {
+  @Prop() readonly block!: ISelectOneResponseBlock
+
+  get contactPropertyName(): string | undefined {
+    return this.block.config.set_contact_property?.[0].property_key
+  }
+
+  get contactProperty(): IContactPropertyOption | null {
+    return find(this.subscriberPropertyFields, contactProperty => contactProperty.name === this.contactPropertyName) ?? null
+  }
+
+  get isMultipleChoiceProperty(): boolean {
+    return this.contactProperty?.data_type === 'multiple_choice'
+  }
+
+  get isTextProperty(): boolean {
+    return this.contactProperty?.data_type === 'text'
+  }
+
+  get isNumberProperty(): boolean {
+    return this.contactProperty?.data_type === 'number'
+  }
+
+  get choiceKeys(): string[] {
+    return this.block.config.choices.map(choice => choice.name)
+  }
+
+  get choiceValueOptions(): IContactPropertyMultipleChoice[] {
+    return this.contactProperty?.choices ?? []
+  }
+
+  getChoiceValue(choiceKey: string): string | number | undefined {
+    return this.block.vendor_metadata?.floip?.ui_metadata?.set_contact_property[0].property_value_mapping?.[choiceKey]
```

Comment:
why adding `[0]` in `set_contact_property` instead of just `set_contact_property`? We don't support multiple contact properties.

Reply:
[Fixed](https://github.com/FLOIP/flow-builder/pull/302/commits/c3f6767d39ade1ac5161c0d0723b0fdd8d6f644b)